src/manifestUtils.ts

```diff
@@ -145,3 +146,29 @@ function isValidManifest(manifest: ManifestSchema): boolean {
     const hasModules = manifest.modules !== undefined
     return hasId && hasModules
 }
+
+/**
+ * Check for runtimes specified in the manifest but are not installed.
+ * @param manifest Manifest to check
+ * @returns List of runtimes that are not installed
+ */
+export function checkForMissingRuntimes(manifest: Manifest): string[] {
+    const runtimeVersion = manifest.manifest['runtime-version']
+    const missingRuntimes = new Set([manifest.manifest.runtime, manifest.manifest.sdk])
+    const missingSdkExtensions = new Set(manifest.manifest['sdk-extensions'])
+
+    for (const availableRuntime of getAvailableRuntimes()) {
+        if (runtimeVersion === availableRuntime.version) {
+            missingRuntimes.delete(availableRuntime.id)
+            continue
+        }
+
+        missingSdkExtensions.delete(availableRuntime.id)
```

Comment:
As mentioned in the comment, this won't work

Reply:
I don't quite understand it. The code only checks if the SDK extension of any version is installed (doesn't check if the version also matches), isn't that less strict?

Reply:
> I don't quite understand it. The code only checks if the SDK extension of any version is installed (doesn't check if the version also matches), isn't that less strict?

Well because it would allow a build even if it shouldn't because you have a wrong version installed. So I would rather not do the check at all for the Sdk extensions and do that in a separate PR in the future. 

> It is only strict on both id and version in checking the SDK itself or the runtime. (I'm not sure though if it is correct to assume that the SDK and runtime should have the same version)

That is fine, using a different SDK and runtime could normally work but you could easily end up in a not working situation because of an ABI change. so definitely not recommended

Reply:
> Well because it would allow a build even if it shouldn't because you have a wrong version installed. So I would rather not do the check at all for the Sdk extensions and do that in a separate PR in the future.

With and without the version check, it would still allow building. I think it would be a bit better if it at least check if the user misspelled the SdkExtension (just don't advertise that it also checks the Sdk extension).

Reply:
I would at least add a FIXME to properly get the required versions for sdk extensions in the future